### ISSUE TITLE
[Bug Report] v-btn-toggle:  Error in beforeDestroy hook: 'TypeError: Cannot read property 'value' of undefined'

### ISSUE THREAD

**Issue by sindrepm** [vuetifyjs/vuetify#3136]
### Versions and Environment
**Vuetify:** 1.0.0-beta.5
**Vue:** 2.5.13
**Browsers:** Chrome 64.0.3282.119
**OS:** Windows 7

### Steps to reproduce
1. Add a `v-btn-toggle` with `mandatory` prop
2. Set the v-model to the value of the first button
3. Do something that triggers destroy of the component (e.g. router navigation, v-if conditional..)

See codepen.

### Expected Behavior
No errors thrown

### Actual Behavior
The following error is thrown on the console:

```
"[Vue warn]: Error in beforeDestroy hook: 'TypeError: Cannot read property 'value' of undefined'

found in

---> <VBtn>
       <VBtnToggle>
         <VContent>
           <VApp>
             <Root>"
```

### Reproduction Link
[https://codepen.io/anon/pen/oEgxGr](https://codepen.io/anon/pen/oEgxGr)

### Other comments
From what I could find, it only happens when `mandatory` prop is used and the `value` assigned to the first button is set as the initial `v-model` value.

**Comment by KaelWD** [comment]
Listeners are bound to the button index:
https://github.com/vuetifyjs/vuetify/blob/251fe7bb445ce75a6a5ddaf3abf3785ce7b11482/src/mixins/button-group.js#L49-L53

Then called after splicing the array:
https://github.com/vuetifyjs/vuetify/blob/251fe7bb445ce75a6a5ddaf3abf3785ce7b11482/src/mixins/button-group.js#L64-L73

Not 100% sure what's going on here, possibly related to #2453.

**Comment by KaelWD** [comment]
Resolved with #3259
